File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/ErrorCodeDescription.java
/*******************************************************************************
 * This file is part of BOINC.
 * http://boinc.berkeley.edu
 * Copyright (C) 2019 University of California
 *
 * BOINC is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License
 * as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * BOINC is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
 ******************************************************************************/
package edu.berkeley.boinc.utils;

import android.os.Parcel;
import android.os.Parcelable;

public class ErrorCodeDescription implements Parcelable {
    public int code;
    public String description;

    public static final Parcelable.Creator<ErrorCodeDescription> CREATOR = new Parcelable.Creator<ErrorCodeDescription>() {
        public ErrorCodeDescription createFromParcel(Parcel in) {
            return new ErrorCodeDescription(in);
        }

        public ErrorCodeDescription[] newArray(int size) {
            return null;
        }
    };

    public ErrorCodeDescription() {
        code = 0;
        description = "";
    }

    public ErrorCodeDescription(Parcel parcel)
    {
        readFromParcel(parcel);
    }

    public ErrorCodeDescription (int _code, String _description) {
        code = _code;
        description = _description;
    }
    public ErrorCodeDescription (int _code) {
        code = _code;
        description = "";
    }

    public int describeContents() {
        return 0;
    }

    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(code);
        parcel.writeString(description);
    }

    public void readFromParcel(Parcel in) {
        code = in.readInt();
        description= in.readString();
    }
}

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/ErrorCodeDescription.kt
/**
 * This file is part of BOINC.
 * http://boinc.berkeley.edu
 * Copyright (C) 2020 University of California
 *
 * BOINC is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License
 * as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * BOINC is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with BOINC.  If not, see <http:></http:>//www.gnu.org/licenses/>.
 */
package edu.berkeley.boinc.utils

import android.os.Parcel
import android.os.Parcelable

data class ErrorCodeDescription @JvmOverloads constructor(val code: Int = 0, val description: String? = "")
    : Parcelable {
    private constructor(parcel: Parcel) : this(parcel.readInt(), parcel.readString())

    val isOK: Boolean
        get() = code == ERR_OK && description.isNullOrEmpty()

    override fun describeContents() = 0

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(code)
        parcel.writeString(description)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<ErrorCodeDescription> = object : Parcelable.Creator<ErrorCodeDescription> {
            override fun createFromParcel(parcel: Parcel) = ErrorCodeDescription(parcel)

            override fun newArray(size: Int) = arrayOfNulls<ErrorCodeDescription>(size)
        }
    }
}
